From 3ebb30a65e2b40610301e5feedda0408ac9f3631 Mon Sep 17 00:00:00 2001
From: James Chapman <jachapma@redhat.com>
Date: Tue, 10 Feb 2026 10:35:48 +0000
Subject: [PATCH] Issue 7230 - Regression in healtcheck NssCheck (#7235)

Description:
Dynamic Certificate lib389 updadates modified get_cert_details() to
return a dict instead of tuple format. _lint_certificate_expiration() and
tls.list_cas() still assumes tuple style access.

Fix:
Update method to use dict key.

Fixes: https://github.com/389ds/389-ds-base/issues/7230

Co-authored-by: @flo-renaud

Reviewed by: @droideck (Thank you)
---
 src/lib389/lib389/cli_ctl/tls.py | 4 ++--
 src/lib389/lib389/nss_ssl.py     | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/lib389/lib389/cli_ctl/tls.py b/src/lib389/lib389/cli_ctl/tls.py
index 8834f5758..91f1f3a59 100644
--- a/src/lib389/lib389/cli_ctl/tls.py
+++ b/src/lib389/lib389/cli_ctl/tls.py
@@ -25,9 +25,9 @@ def list_client_cas(inst, log, args):
 
 def list_cas(inst, log, args):
     tls = NssSsl(dirsrv=inst)
-    # This turns an array of [('CA', 'C,,')]
+    # Returns a list of cert dicts, eg {'cn': 'nickname', etc}
     for c in tls.list_ca_certs():
-        log.info(c[0])
+        log.info(c['cn'])
 
 
 def show_cert(inst, log, args):
diff --git a/src/lib389/lib389/nss_ssl.py b/src/lib389/lib389/nss_ssl.py
index 764434166..fae65d19c 100644
--- a/src/lib389/lib389/nss_ssl.py
+++ b/src/lib389/lib389/nss_ssl.py
@@ -91,13 +91,13 @@ class NssSsl(DSLint):
             if diff_date < timedelta(days=0):
                 # Expired
                 report = copy.deepcopy(DSCERTLE0002)
-                report['detail'] = report['detail'].replace('CERT', cert[0])
+                report['detail'] = report['detail'].replace('CERT', cert['cn'])
                 report['check'] = f'tls:certificate_expiration'
                 yield report
             elif diff_date < timedelta(days=30):
                 # Expiring within 30 days
                 report = copy.deepcopy(DSCERTLE0001)
-                report['detail'] = report['detail'].replace('CERT', cert[0])
+                report['detail'] = report['detail'].replace('CERT', cert['cn'])
                 report['check'] = f'tls:certificate_expiration'
                 yield report
 
-- 
2.52.0

